### PR TITLE
Added lookup aggregates builder

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1936

### CODE REVIEW

File: driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy [hunk 1]
```diff
@@ -189,4 +191,32 @@ class AggregatesFunctionalSpecification extends OperationFunctionalSpecification
...
     }
 
+
+    @IgnoreIf({ !serverVersionAtLeast(asList(3, 1, 8)) || !isEnterpriseServer() })
+    def '$lookup'() {
+        given:
+        def fromCollectionName = 'lookupCollection'
+        def fromHelper = getCollectionHelper(new MongoNamespace(getDatabaseName(), fromCollectionName))
+
+        getCollectionHelper().drop()
```
Comment (jyemin) [inline driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy]: This gets dropped in base class setup, so no need to do it here as well I think.

**Comment by jyemin** [comment]
lgtm, but have Craig give it a quick once-over too.

**Comment by jyemin** [comment]
Just realized I didn't say a reference docs addition for this.  Currently there's a section for every stage in aggregation.md

File: driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy (hunk 1 above)
Comment (rozza) [inline driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy]: This is needed to as setup inserts documents.

**Comment by rozza** [comment]
Rebased with master - added documentation.

File: driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy (hunk 1 above)
Comment (jyemin) [inline driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy]: Acknowledged.

**Comment by rozza** [comment]
Done.

**Comment by jyemin** [comment]
lgtm

**Comment by rozza** [comment]
Rebased and merged

**Pull request closed without merging**